spotinst/resource_spotinst_ocean_aks_np_virtual_node_group_test.go

```diff
@@ -0,0 +1,627 @@
+			return err
+		}
+		if spotinst.StringValue(resp.VirtualNodeGroup.ID) != rs.Primary.Attributes["id"] {
+			return fmt.Errorf("Virtual Node Group not found: %+v,\n %+v\n", resp.VirtualNodeGroup, rs.Primary.Attributes)
+		}
+		*virtualNodeGroup = *resp.VirtualNodeGroup
+		return nil
+	}
+}
+
+type AKSNPVirtualNodeGroupConfigMetadata struct {
+	provider             string
+	vngResourceName      string
+	updateBaselineFields bool
+}
+
+func createOceanAKSNPVirtualNodeGroupTerraform(vngcm *AKSNPVirtualNodeGroupConfigMetadata, formatToUse string) string {
+	if vngcm == nil {
+		return ""
+	}
+
+	if vngcm.provider == "" {
+		vngcm.provider = "azure"
+	}
+
+	template :=
+		`provider "azure" {
+	 token   = "fake"
+	 account = "fake"
+	}
+	`
+
+	format := formatToUse
+
+	if vngcm.updateBaselineFields {
+		template += fmt.Sprintf(format,
+			vngcm.vngResourceName,
+			vngcm.provider,
+		)
+	} else {
+		template += fmt.Sprintf(format,
+			vngcm.vngResourceName,
+			vngcm.provider,
+		)
+	}
+
+	log.Printf("Terraform LaunchSpec template:\n%v", template)
+
+	return template
+}
+
+// region OceanAKSNPVirtualNodeGroup: Baseline
+func TestAccSpotinstOceanAKSNPVirtualNodeGroup_Baseline(t *testing.T) {
+	vngResourceName := "test-aks-vng"
+	resourceName := createOceanAKSNPVirtualNodeGroupResource(vngResourceName)
+
+	var virtualNodeGroup azure_np.VirtualNodeGroup
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t, "azure") },
+		Providers:    TestAccProviders,
+		CheckDestroy: testOceanAKSNPVirtualNodeGroupDestroy,
+
+		Steps: []resource.TestStep{
+			{
+				Config: createOceanAKSNPVirtualNodeGroupTerraform(&AKSNPVirtualNodeGroupConfigMetadata{vngResourceName: vngResourceName, updateBaselineFields: true}, testBaselineOceanAKSNPVirtualNodeGroup_Create),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckOceanAKSNPVirtualNodeGroupExists(&virtualNodeGroup, resourceName),
+					resource.TestCheckResourceAttr(resourceName, "name", "testVng"),
+					resource.TestCheckResourceAttr(resourceName, "ocean_id", "o-751eaa33"),
+					resource.TestCheckResourceAttr(resourceName, "min_count", "1"),
+					resource.TestCheckResourceAttr(resourceName, "max_count", "100"),
+					resource.TestCheckResourceAttr(resourceName, "max_pods_per_node", "30"),
+					resource.TestCheckResourceAttr(resourceName, "enable_node_public_ip", "true"),
+					resource.TestCheckResourceAttr(resourceName, "os_disk_size_gb", "32"),
+					resource.TestCheckResourceAttr(resourceName, "os_disk_type", "Managed"),
+					resource.TestCheckResourceAttr(resourceName, "os_type", "Linux"),
+					resource.TestCheckResourceAttr(resourceName, "os_sku", "Ubuntu"),
+					resource.TestCheckResourceAttr(resourceName, "kubernetes_version", "1.26"),
+					resource.TestCheckResourceAttr(resourceName, "spot_percentage", "50"),
+					resource.TestCheckResourceAttr(resourceName, "fallback_to_ondemand", "false"),
+					//resource.TestCheckResourceAttr(resourceName, "pod_subnet_ids.#", "1"),
```

Comment:
what is the reason for commenting pod_subnet_ids

Reply:
we get error when we use both pod subnet ids and vnet ids